src/query-creator.ts

```diff
@@ -464,41 +427,9 @@ export class QueryCreator<DB> {
    * console.log(result.numUpdatedRows)
    * ```
    */
-  updateTable<TE extends keyof DB & string>(
-    from: TE[],
-  ): UpdateQueryBuilder<
-    DB,
-    ExtractTableAlias<DB, TE>,
-    ExtractTableAlias<DB, TE>,
-    UpdateResult
-  >
-
-  updateTable<TE extends TableExpression<DB, never>>(
-    from: TE[],
-  ): UpdateQueryBuilder<
-    From<DB, TE>,
-    FromTables<DB, never, TE>,
-    FromTables<DB, never, TE>,
-    UpdateResult
-  >
-
-  updateTable<TE extends keyof DB & string>(
+  updateTable<TE extends TableExpressionOrList<DB, never>>(
     from: TE,
-  ): UpdateQueryBuilder<
-    DB,
-    ExtractTableAlias<DB, TE>,
-    ExtractTableAlias<DB, TE>,
-    UpdateResult
-  >
-
-  updateTable<TE extends AnyAliasedTable<DB>>(
-    from: TE,
-  ): UpdateQueryBuilder<
-    DB & PickTableWithAlias<DB, TE>,
-    ExtractTableAlias<DB & PickTableWithAlias<DB, TE>, TE>,
-    ExtractTableAlias<DB & PickTableWithAlias<DB, TE>, TE>,
-    UpdateResult
-  >
+  ): UpdateTable<DB, TE>
 
   updateTable(tables: TableExpressionOrList<any, any>): any {
```

Comment:
do we still need the split between declaration and implementation?

Reply:
Not needed specifically, but the other option is to have `as any` casts inside the implementation. I think that's still better so I'll combine them.